src/System.Runtime/tests/System/String.Split.cs

```diff
@@ -0,0 +1,515 @@
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(new[] { ',' }, count));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(new[] { ',' }, count, options));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(",", count, options));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitInvalidOptions()
+    {
+        const string value = "a,b";
+        const int count = int.MaxValue;
+        const StringSplitOptions optionsTooLow = StringSplitOptions.None - 1;
+        const StringSplitOptions optionsTooHigh = StringSplitOptions.RemoveEmptyEntries + 1;
+
+        Assert.Throws<ArgumentException>(() => value.Split(',', optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(',', optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(',', count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(',', count, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, count, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(",", optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(",", optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(",", count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(",", count, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, count, optionsTooHigh));
+    }
+
+    [Fact]
+    public static void TestSplitZeroCountEmptyResult()
+    {
+        const string value = "a,b";
+        const int count = 0;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        string[] expected = new string[0];
+
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitEmptyValueWithRemoveEmptyEntriesOptionEmptyResult()
```

Comment:
This is the same test as the InlineData test on line 171. Should we remove the data-driven test or this one?